x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/fleet/agent_list_page/index.tsx

```diff
@@ -47,20 +59,101 @@ const Divider = styled.div`
 
 const REFRESH_INTERVAL_MS = 5000;
 
+const statusFilters = [
+  {
+    status: 'online',
+    label: i18n.translate('xpack.ingestManager.agentList.statusOnlineFilterText', {
+      defaultMessage: 'Online',
+    }),
+  },
+  {
+    status: 'offline',
+    label: i18n.translate('xpack.ingestManager.agentList.statusOfflineFilterText', {
+      defaultMessage: 'Offline',
+    }),
+  },
+  ,
+  {
+    status: 'error',
+    label: i18n.translate('xpack.ingestManager.agentList.statusErrorFilterText', {
+      defaultMessage: 'Error',
+    }),
+  },
+] as Array<{ label: string; status: string }>;
+
+const RowActions = React.memo<{ agent: Agent; refresh: () => void }>(({ agent, refresh }) => {
+  const DETAILS_URI = useLink(FLEET_AGENT_DETAIL_PATH);
+  const [isOpen, setIsOpen] = useState(false);
+  const handleCloseMenu = useCallback(() => setIsOpen(false), [setIsOpen]);
+  const handleToggleMenu = useCallback(() => setIsOpen(!isOpen), [isOpen]);
+
+  return (
+    <EuiPopover
+      anchorPosition="downRight"
+      panelPaddingSize="none"
+      button={
+        <EuiButtonIcon
+          iconType="boxesHorizontal"
+          onClick={handleToggleMenu}
+          aria-label={i18n.translate('xpack.ingestManager.agentList.actionsMenuText', {
+            defaultMessage: 'Open',
+          })}
+        />
+      }
+      isOpen={isOpen}
+      closePopover={handleCloseMenu}
+    >
+      <EuiContextMenuPanel
+        items={[
+          <EuiContextMenuItem icon="inspect" href={`${DETAILS_URI}${agent.id}`} key="viewConfig">
+            <FormattedMessage
+              id="xpack.ingestManager.agentList.viewActionText"
+              defaultMessage="View Agent"
+            />
+          </EuiContextMenuItem>,
+
+          <AgentUnenrollProvider>
+            {unenrollAgentsPrompt => (
+              <EuiContextMenuItem
+                color="danger"
+                icon="cross"
+                onClick={() => {
+                  unenrollAgentsPrompt([agent.id], 1, () => {
+                    refresh();
+                  });
+                }}
+              >
+                <FormattedMessage
+                  id="xpack.ingestManager.agentList.unenrollOneButton"
+                  defaultMessage="unenroll"
+                />
+              </EuiContextMenuItem>
+            )}
+          </AgentUnenrollProvider>,
+        ]}
+      />
+    </EuiPopover>
+  );
+});
+
 export const AgentListPage: React.FunctionComponent<{}> = () => {
+  const defaultKuery: string = (useUrlParams().kuery as string) || '';
```

Comment:
FYI: In the commit I just merged, I changed the interface of `useUrlParam()`. It now returns an object that contains properties `urlParams` (object with url params) and `toUrlParams` (a function that allows you to stringify an object to url params)